system-test/resource.js

```diff
@@ -100,15 +58,73 @@ describe('Resource', function() {
         done();
       });
     });
+  });
+
+  // Auth through the gcloud SDK is required to:
+  //
+  //   - Create a project
+  //   - Set metadata
+  //   - Restore a project
+  //   - Delete a project
+  describe('lifecycle', function() {
+    var CAN_RUN_TESTS = true;
+
+    var resource = new Resource({
+      projectId: env.projectId
+    });
+
+    var PROJECT_ID = 'gcloud-tests-' + Date.now();
+    var project = resource.project(PROJECT_ID);
+
+    before(function(done) {
+      var authClient = googleAuth();
+
+      // See if an auth token exists.
+      authClient.getToken(function(err) {
+        if (err) {
+          if (err.code === 400) {
+            CAN_RUN_TESTS = false;
+            done();
+          } else {
+            done(err);
+          }
+          return;
+        }
+
+        project.create(done);
+      });
+    });
+
+    beforeEach(function() {
+      if (!CAN_RUN_TESTS) {
+        this.skip();
+      }
+    });
+
+    after(function(done) {
+      if (!CAN_RUN_TESTS) {
+        this.skip();
+        return;
+      }
+
+      project.delete(done);
+    });
+
+    it('should have created the project', function() {
+      assert.strictEqual(project.metadata.projectId, PROJECT_ID);
+    });
+
+    it('should set metadata', function(done) {
```

Comment:
This is consistently failing for me locally with an `Uncaught ApiError: Not Found`